src/API/Site/Controllers/GTINMigrationController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\MigrateGTIN;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Exception;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class GTINMigrationController offering API endpoint for GTIN field Migration
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers
+ */
+class GTINMigrationController extends BaseController {
+	use EmptySchemaPropertiesTrait;
+
+	/**
+	 * Job responsible to run the migration in the background.
+	 *
+	 * @var MigrateGTIN
+	 */
+	protected $job;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param RESTServer  $server
+	 * @param MigrateGTIN $job
+	 */
+	public function __construct( RESTServer $server, MigrateGTIN $job ) {
+		parent::__construct( $server );
+		$this->job = $job;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'gtin-migration-start',
+			[
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->start_migration_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_schema_properties(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+
+	/**
+	 * Callback function for scheduling GTIN migration job.
+	 *
+	 * @return callable
+	 */
+	protected function start_migration_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				if ( ! $this->job->can_schedule( [ 1 ] ) ) {
+					return new Response(
+						[
+							'status'  => 'error',
+							'message' => __( 'GTIN Migration cannot be scheduled.', 'google-listings-and-ads' ),
```

Comment:
I was thinking about block when it is completed as well. My main question was about if a user wanted to run again the migration after its completed. This case seems to be not very usual as the GTIN will be read-only in the extension tabs. So no chance for the user to add new GTINS to migrate. 

So do you suggest to block it after it's completed? The banner won't show tho... so I don't think its a problem to keep it as it is.

Reply:
I don't suggest to block it, in case it's useful in troubleshooting to re-trigger.
Just wanted to make a note of the behaviour, just to confirm it's expected.